lib/oli_web/live/workspaces/utils.ex

```diff
@@ -213,6 +216,13 @@ defmodule OliWeb.Workspaces.Utils do
     end
   end
 
+  def is_navigate_route(route) do
```

Comment:
I'm not sure what this function does.  Can you add some documentation for it, as well as include a unit test?

Reply:
This should be a private function. Its purpose is to identify the type of navigation needed for each route: either `href` or `navigate`. This identification is then used to set the appropriate navigation logic. Although it's not the most elegant approach, as it disrupts our smooth navigation between live pages within the same live session, it's necessary. Without this distinction, we won't load the required JS files to run the React components on the view.

In researching this issue, I found a similar case in this post: https://elixirforum.com/t/is-it-possible-to-have-page-specific-javascript-with-liveview/32614. I tried a few experiments based on this but couldn’t get it to work, so I’ve added it to my to-do list.

This issue also affects our breadcrumbs, as note [here](https://github.com/Simon-Initiative/oli-torus/pull/5174/files#r1790755239).

Reply:
Fran, as we discussed the other day, a way to get the scripts loaded in an async way is reusing the already existing `LoadSurveyScripts` hook (there are some references in the code where we use this strategy). I am almost sure this approach will work but, if it doesn't, what do you think about moving this page out of the live_session? In this way you will "force" to fully reload the page (and scripts should load from scratch), and will remove the need of this `is_navigate_route` helper function.